File: leetCode/fukun/lc_101.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const isSymmetric = function (root) {
    const isSymmetric = function (p, q) {
        if (!p && !q) return true;
        if (!q || !p || q.val !== p.val) return false;
        return isSymmetric(p.left, q.right) && isSymmetric(p.right, q.left);
    };
    return root ? isSymmetric(root.left, root.right) : true;
};

let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(2);

console.log(isSymmetric(t));

File: leetCode/lwc/lc_101.java
package solution2;

/**
 * @author lwc
 * @date 2019/10/27 11:39
 */
public class U101 {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        return recursive(root.left,root.right);
    }

    public boolean recursive(TreeNode p, TreeNode q){
        if(p == null && q == null) return true;
        if(p == null || q==null || p.val != q.val) return false;
        return recursive(p.left,q.right) && recursive(p.right,q.left);
    }
}
